server/server-netty/src/main/java/software/amazon/smithy/java/server/netty/HttpRequestHandler.java

```diff
@@ -98,6 +104,7 @@ private void writeResponse(Channel channel, HttpJob job) {
             if (serializedValue.contentType() != null) {
                 response.headers().set("content-type", serializedValue.contentType());
             }
+            CorsHeaders.of(job).forEach(response.headers()::set);
```

Comment:
`response.headers().putHeaders(CorsHeaders.of(job));` would be better. The current method requires allocating and invoking a state-capturing lambda, which is surprisingly hard for the JIT to inline and optimize away.

The best option, though, is to set the headers in `CorsHeaders#of`.